src/client/clientlauncher.cpp

```diff
@@ -127,7 +130,30 @@ bool ClientLauncher::run(GameParams &game_params, const Settings &cmd_args)
 	skin->setColor(gui::EGDC_3D_SHADOW, video::SColor(255, 0, 0, 0));
 	skin->setColor(gui::EGDC_HIGH_LIGHT, video::SColor(255, 70, 120, 50));
 	skin->setColor(gui::EGDC_HIGH_LIGHT_TEXT, video::SColor(255, 255, 255, 255));
-
+#ifdef __ANDROID__
+	float density = porting::getDisplayDensity();
+	skin->setSize(gui::EGDS_CHECK_BOX_WIDTH, (s32)(17.0f * density));
+	skin->setSize(gui::EGDS_SCROLLBAR_SIZE, (s32)(15.0f * density));
+	skin->setSize(gui::EGDS_WINDOW_BUTTON_WIDTH, (s32)(15.0f * density));
+	if (density > 1.5f) {
+		std::string sprite_path = porting::path_user + "/textures/base/pack/";
+		if (density > 3.5f)
+			sprite_path += "checkbox_64.png";
+		else if (density > 2.0f)
+			sprite_path += "checkbox_32.png";
+		else
+			sprite_path += "checkbox_16.png";
+		// Texture dimensions should be a power of 2
+		gui::IGUISpriteBank *sprites = skin->getSpriteBank();
+		video::IVideoDriver *driver = RenderingEngine::get_video_driver();
+		video::ITexture *sprite_texture = driver->getTexture(sprite_path.c_str());
```

Comment:
you can inline that in the conditional directly

Reply:
> you can inline that in the conditional directly

Sorry, I have no idea what you mean by this. Please show me how you would like it.

Reply:
I think he meant it like this:
```C++
if (video::ITexture *sprite_texture = driver->getTexture(sprite_path.c_str())) {
	// Use sprite_texture
```
This doesn't look great either, so you may just keep it as-is.

Reply:
Ah, I see he means now but surely it looks better like this.